tools/ds-identify

```diff
@@ -763,6 +769,51 @@ check_writable_seed_dir() {
     check_seed_dir "$@"
 }
 
+debugts() {
+    local lvl="$1"
+    shift
+    [ "$lvl" -gt "${DI_DEBUG_LEVEL}" ] && return
+    read_uptime
+    debug "$lvl" "[up ${_RET}s]" "$@"
+}
+
+probe_cdrom() {
+    cached "${STATE_CDROM_PROBED}" && return "${STATE_CDROM_PROBED}"
+    if [ -z "$PROBE_CDROM_MODULES" ]; then
+        STATE_CDROM_PROBED=1
+        return 0
+    fi
+
+    local fpath=/dev/cdrom
+    if [ -b "$fpath" ]; then
+        STATE_CDROM_PROBED=1;
+        debug 1 "$fpath existed before probe_cdrom"
+        return 0;
+    fi
+
+    debugts 1 "attempting modprobe of ${PROBE_CDROM_MODULES}"
+    local m="" modfails=""
+    for m in ${PROBE_CDROM_MODULES}; do
+        modprobe -b "$m" >/dev/null 2>&1 || modfails="${modfails:+$modfails }$m:$?"
+    done
+    [ -z "$modfails" ] ||
+        debug 1 "probe_cdrom modprobe had unsuccessful attempts: $modfails"
+
+    # Some Linux distros/non-Linux OSes may not have udev
+    if command -v udevadm >/dev/null 2>&1; then
+        debugts 1 "udev settling for $fpath"
+        if ! udevadm settle "--exit-if-exists=$fpath"; then
```

Comment:
This solution looks incorrect. To my knowledge the kernel makes no guarantees that loading a module will cause an associated uevent to be queued before the driver is done loading. This might work sometimes, but I don't think that there are any guarantees that it will always work. Am I missing something?

Also, generators are supposed to be fast and this operation is functionally a wait.